docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java

```diff
@@ -152,8 +161,9 @@ private void slaveShutdown(TaskListener listener) throws DockerException, IOExce
             String tagToken = getAdditionalTag(listener);
 
             if( !Strings.isNullOrEmpty(tagToken) ) {
-                // ?? client.image(tag_image).tag(tagToken, false);
-                client.tagImageCmd(tag_image,null,tagToken).exec();
+                client.tagImageCmd(imageId, imageName.toString(), tagToken)
```

Comment:
If you created a real wrapper object, then using .toString() is bad idea, wrapper class should provide some clear getter that wouldn't depend on toString() implementation that usually has format "Key = value" for printing all object internals.